File: Python/010.py
# The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.
# Find the sum of all the primes below two million.

import math


def prime_sieve(n):
    """Use the Sieve of Eratosthenes to produce all prime numbers from 2 through n.
    Adapated from the wikipedia pseudocode."""

    sieve = [True] * n

    for i in range(2, int(math.sqrt(n)) + 1):
        if sieve[i]:
            for j in range(2 * i, n, i):
                sieve[j] = False

    return [x for x in range(2, n) if sieve[x]]


if __name__ == "__main__":
    bound = 2000000
    primes = prime_sieve(bound)
    print("Sum of primes below two million", sum(primes))

File: Java/010.java
import java.util.Arrays;

public class Problem010 {

    /**
     * Use the Sieve of Eratosthenes to produce all prime numbers from 2 through n.
     * Adapted from the wikipedia pseudocode.
     */
    public static long primeSieve(int n) {
        boolean[] sieve = new boolean[n];
        Arrays.fill(sieve, true);

        for (int i = 2; i * i < n; i++) {
            if (sieve[i]) {
                for (int j = i * i; j < n; j += i) {
                    sieve[j] = false;
                }
            }
        }

        long sum = 0;
        for (int i = 2; i < n; i++) {
            if (sieve[i]) {
                sum += i;
            }
        }
        return sum;
    }

    public static void main(String[] args) {
        int bound = 2_000_000;
        long sum = primeSieve(bound);
        System.out.println("Sum of primes below two million: " + sum);
    }
}
